### PR TITLE
#495: add missing hooks in git-meta.

### PR DESCRIPTION
**Pull request by s-xiang**

### CODE REVIEW

File: node/lib/cmd/rebase.js [hunk 1]
```diff
@@ -110,4 +111,7 @@ exports.executeableSubcommand = co.wrap(function *(args) {
         const commit = yield repo.getCommit(committish.id());
         yield RebaseUtil.rebase(repo, commit);
     }
+
+    // Run post-rewrite hook with "rebase" as arg, means rebase command invoked this hook.
+    yield Hook.execHook("post-rewrite", args=["rebase"]);
```
Comment (novalis) [inline node/lib/cmd/rebase.js]: s/args=//

Comment (novalis) [reply node/lib/cmd/rebase.js]: (it's not doing what you think it's doing)

File: node/lib/util/hook.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ */
+exports.execHook = co.wrap(function *(name, args=[]) {
+    assert.isString(name);
+
+    const hookPath = path.join(GitUtil.getRootGitDirectory(), ".git/hooks");
+    const absPath = path.resolve(hookPath, name);
+
+    try {
+        const result = yield ChildProcess.execFile(absPath, args);
+        return result.stdout;
```
Comment (bpeabody) [inline node/lib/util/hook.js]: If this method has a return value, please document it above.

Comment (bpeabody) [reply node/lib/util/hook.js]: Explicitly call out the arguments in the documentation and list their names in the jsduck like:
```
@param {String} name ...
@param {String[]} args ...
```

**Pull request merged**